### PR TITLE
Determinant appendix

### PR DESCRIPTION
**Pull request by Xintong-Li**
Appendix: determinant (draft)

This PR is about the determinant appendix.

Some Problems:
1. To illustrate the diagonal method, I used the `xymatrix` package. However, this package interally override `\Col` and `\Row`. To compensate for this, I included this package at the beginning of appendix 3 (resulting in a "can only be used in preample" error).

2. To illustrate the cofactor method, I used the `nicematrix` package. I included this package in the `preample.tex`

Update: I used `nicematrix` to recreate the diagonal trick part so that `xymatrix` is no longer needed.

### CODE REVIEW

**Comment by siefkenj** [comment]
Run your code through https://siefkenj.github.io/latex-parser/   It'll make the formatting nice.

File: book/modules/appendix3.tex [hunk 1]
```diff
@@ -0,0 +1,172 @@
+Determinant is useful: it measures the volume distortion of a linear transformation, it helps us determine if a matrix is invertible of not, and it provides us with a simple formula for solving \(A\vec x=\vec y\) when \(\vec x\) is invertible.
```
Comment (siefkenj) [inline book/modules/appendix3.tex:1]: `x` is invertible?

File: book/modules/appendix3.tex [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+
+In Module 14, you will learn an algorithm for evaluating determinants via row reduction. However, for \(2\times 2\) and \(3\times 3\) determinants, we have methods which are more convenient than row reduction. We will explore these methods in this appendix.
```
Comment (siefkenj) [inline book/modules/appendix3.tex:3]: Add a `\label` to the start of Module 14 (right after the `\Title`) and then use a `\ref`. No absolute references allowed.

Comment (siefkenj) [reply book/modules/appendix3.tex:3]: Keep a consistent tense: "Module 14 discusses ..."

Comment (siefkenj) [reply book/modules/appendix3.tex:3]: They aren't more convenient. They are direct formulas.

File: book/modules/appendix3.tex [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+
+\Heading{A formula for \(2\times 2\) determinants}
+For a \(2\times 2\) matrix, once we know its four entries, we can calculate its determinant directly.
+\begin{theorem}[\(2\times 2\) determinant formula]
+  Let \(M\) be a \(2\times 2\) matrix. Suppose \(M=\mat{a & b \\ c & d}\). Then \[\det(M)=ad-bc.\]
```
Comment (siefkenj) [inline book/modules/appendix3.tex:8]: Let `M=\mat{...}`. Then, ...

File: book/modules/appendix3.tex [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+\end{theorem}
+\begin{example}
+Use the \(2\times 2\) determinant formula to find \(\det(M)\), where \(M=\mat{1 & 6 \\ 2  & 7}\).
+\end{example}
+\textbf{Solution. }Apply the formula, and we get \[\det(M)=1\times 7-2\times 6=-5.\]
```
Comment (siefkenj) [inline book/modules/appendix3.tex:13]: Solution goes inside the example. Don't prefix the solution with `Solution.`

File: book/modules/appendix3.tex [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+Now we turn to \(3\times 3\) determinants. We still have a manageable formula, although it is less appealing than the \(2\times 2\) formula:
+\begin{theorem}[\(3\times 3\) determinant formula]
+  Let \(M\) be a \(3\times 3\) matrix. Suppose \(M=\mat{a & b & c \\ d & e & f \\ g & h & i}\). Then \[\det(M)=aei+bfg+cdh-gec-hfa-idb.\]
+\end{theorem}
+You see, this formula is not easy to remember. To help you remember this formula, here's a trick called the \textit{diagonal trick}. Applying this formula, then, is no more than applying this trick on the matrix.
+
+Now we outline the procedure of calculating \(\det(\mat{a & b & c \\ d & e & f \\ g & h & i})\) via this trick.
+
+\textbf{Step 1. }Copy the matrix, and then copy the \emph{first two columns} to the right of the matrix:
+\[\begin{NiceArray}{CCCCCCCCCCC}[margin]
```
Comment (siefkenj) [inline book/modules/appendix3.tex:29]: Why don't you do this with the augmented matrix syntax? Then you can use `{\color{...} ...}` in the cells to distinguish the columns/diagonals.

File: book/modules/appendix3.tex [hunk 6]
```diff
@@ -0,0 +1,172 @@
...
+\Heading{Calculating \(3\times 3\) determinants (I): The diagonal trick}
+Now we turn to \(3\times 3\) determinants. We still have a manageable formula, although it is less appealing than the \(2\times 2\) formula:
+\begin{theorem}[\(3\times 3\) determinant formula]
+  Let \(M\) be a \(3\times 3\) matrix. Suppose \(M=\mat{a & b & c \\ d & e & f \\ g & h & i}\). Then \[\det(M)=aei+bfg+cdh-gec-hfa-idb.\]
+\end{theorem}
+You see, this formula is not easy to remember. To help you remember this formula, here's a trick called the \textit{diagonal trick}. Applying this formula, then, is no more than applying this trick on the matrix.
+
+Now we outline the procedure of calculating \(\det(\mat{a & b & c \\ d & e & f \\ g & h & i})\) via this trick.
+
+\textbf{Step 1. }Copy the matrix, and then copy the \emph{first two columns} to the right of the matrix:
```
Comment (siefkenj) [inline book/modules/appendix3.tex:28]: Use an `\begin{itemize}` if you are going to list steps. You can then do `\item[Step 1]`

**Comment by Xintong-Li** [comment]
Update:

1. Used Latex Parser to reformat the source code.

2. Fixed some wording and typographical problems pointed out by previous PR, as well as other typos.

Some problems (to be solved before merging):

1. I am now using `itemize` environment for diagonal trick. However, it results in the following problem:

image

Is there a way to solve it?

2. To illustrate diagonal trick, now I use the augmented matrix syntax. Currently the effect is the as first matrix illustrated below.
Some comments:
(a) Would it be better if I use `vmatrix` instead of `bmatrix`?
(b) For the two hightlighting methods, which is appears to be better?
(I commented the code for the second highlighting methods. It need to be deleted before merging.)

image

**Comment by siefkenj** [comment]
Look at the documentation for the `enumitem` package. It will tell you how to adjust the left margin of an `itemize` to give more space. Also, we should use an `enumerate` instead, since we're actually enumerating.

The background color highlighting looks great! We need to make sure it also works for color blind people, so pick colors that you can tell the difference between in a black-and-white screenshot.

**Comment by Xintong-Li** [comment]
Update highlighting.

Now, the diagonal trick illustrations look like 
image
In black and white, they look like 
image

**Comment by siefkenj** [comment]
Certainly looks nice in black and white. Can you find more pleasing colors, though?  These color scales work well in B&W: https://www.thinkingondata.com/something-about-viridis-library/

File: book/modules/appendix3-exercises.tex [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+		along the first column, but she is quite unconfident about the formula on
+		expanding along an arbitrary column/row. You are going to help her solve
+		her confusion.
+		\begin{enumerate}
+			\item Explain to her, using complete English sentence, why we can expand
+				along the second column. (Hint: you will probably need to use a
+				theorem from Module \ref{determinant module}.)
+
+			\item Inspired by your argument, Kokoro then developed an argument
+				about the invalidity of expanding along the third columns: \textit{If
```
Comment (siefkenj) [inline book/modules/appendix3-exercises.tex:37]: Use `\emph`, not `\textit`

File: book/modules/appendix3.tex [hunk 8]
```diff
@@ -0,0 +1,241 @@
...
+	-2 & 3 & 1 & -2 & 3 \\
+	0 & 2 & 1 & 0 & 2
+	\end{bNiceArray}\]
+	\[
+		G={\color{Salmon}0 \times 3 \times 0}+{\color{SeaGreen}2\times 1 \times 1}+{\color{Sepia} 1 \times (-2)\times 4}=-6.
+	\]
+	Thus, $\det\mat{1 & 4 & 0 \\ -2 & 3 & 1 \\ 0 & 2 & 1}=3-(-6)=9$.
+\end{example}
+
+\Heading{Calculating $3\times 3$ determinants (II): Expanding along the first column}
```
Comment (siefkenj) [inline book/modules/appendix3.tex:125]: Title Capitalization for Headings

**Comment by luema** [comment]
> Update highlighting.
> 
> Now, the diagonal trick illustrations look like
> image
> In black and white, they look like
> image

Just to add to what Jason said about the colour, I've used this site in the past to find colourblind-friendly palettes: https://colorbrewer2.org/#type=qualitative&scheme=Dark2&n=3

File: book/modules/appendix3.tex [hunk 9]
```diff
@@ -0,0 +1,241 @@
+Determinant is useful: it measures the volume distortion of a linear transformation,
+it helps us determine if a matrix is invertible of not, and it provides us with a
+simple formula for solving $A\vec x=\vec y$ when $A$ is invertible.
+
+Module \ref{determinant module} discusses an algorithm for evaluating determinants
+via row reduction. However, for $2\times 2$ and $3\times 3$ determinants, we
+have direct formulas. We will explore these formulas in this appendix.
+
+\Heading{A formula for $2\times 2$ determinants} For a $2\times 2$ matrix, once we
```
Comment (luema) [inline book/modules/appendix3.tex:9]: I think there should be some consistency among the headings. Something like `Computing 2x2 Determinants`, `Computing 3x3 Determinants (I)...`, `Computing 3x3 Determinants (II)...`, etc.

**Comment by Xintong-Li** [comment]
Update: Color and other issues

1. Updated color. The color palette is inspired by _3-class Paired_ palette on https://colorbrewer2.org/#type=qualitative&scheme=Paired&n=3

image
image

2. Reformatted the appearence of determinants. Now arugment of any determinant function always has a pair of parentheses (of suitaible size) surrounding it.

3. Reformatted the appearence of fractions. If a fraction is accompanied by a matrix, then the fraction is set to `displaystyle`.

4. Remove a problem about vector products (because there's already a problem reg. vector products in tutorial). Added new problems: determine $\operatorname{Vol~Change}(\mathcal{T})$ for a given transformation $\mathcal{T}:\mathbf{R}^3\rightarrow\mathbfR}^3$, and determine, using determinant, if $\operatorname{Col}(A)$ has the max possible dimension.

File: book/modules/appendix3.tex [hunk 10]
```diff
@@ -0,0 +1,241 @@
...
+This is the origin of the name ``expanding along the first column''.
+
+If we let $M_{rs}$ be the determinant of the matrix obtained by deleting the $r$'th
+row and $s$'th column, then
+\[
+	\det(A)=a_{11}M_{11}-a_{21}M_{21}+a_{31}M_{31}.
+\]
+Using the summation notation, this can be written in a more compact form:
+\[
+	\det(A)=\sum_{r=1}^{3}(-1)^{r+1}a_{r1}M_{r1}.\footnote{In fact, this can be generalized to determinants of arbitriary size: for any $n
```
Comment (siefkenj) [inline book/modules/appendix3.tex:175]: `.` after footnote.

File: book/modules/appendix3.tex [hunk 11]
```diff
@@ -0,0 +1,241 @@
...
+	\end{bNiceArray}\]
+	\[
+		G={\color{LimeGreen}0 \times 3 \times 0}+{\color{NavyBlue}2\times 1 \times 1}+{\color{CornflowerBlue} 1 \times (-2)\times 4}=-6.
+	\]
+	Thus, $\det\left(\mat{1 & 4 & 0 \\ -2 & 3 & 1 \\ 0 & 2 & 1}\right)=3-(-6)=9$.
+\end{example}
+
+\Heading{Calculating $3\times 3$ Determinants (II): Expanding along the First Column}
+The second method of calculating $3\times 3$ determinants actually uses $2\times
+2$ determinants. This method is called \textbf{expanding along the first column}.
```
Comment (siefkenj) [inline book/modules/appendix3.tex:127]: `textbf` -> `emph`

File: book/modules/appendix3.tex [hunk 12]
```diff
@@ -0,0 +1,241 @@
...
+		= & a_{11}(a_{22}a_{33}-a_{23}a_{32})-a_{21}(a_{12}a_{33}-a_{13}a_{32})+a_{31}(a_{12}a_{23}-a_{13}a_{22})                                                    \\
+		= & a_{11}a_{22}a_{33}+a_{12}a_{23}a_{31}+a_{13}a_{21}a_{32}-a_{31}a_{22}a_{13}-a_{32}a_{23}a_{11}-a_{33}a_{21}a_{12}.
+	\end{align*}
+	This agrees with the $3\times 3$ determinant formula.
+\end{proof}
+
+Before moving on, we give an explanation about the name ``expanding along the first
+column''.
+
+The formula involves three determinants. If we delete the first column and first row of \(A\). we are left with a \(2\times 2\) matrix (below, a highlighted column/row means that this column/row is deleted).
```
Comment (siefkenj) [inline book/modules/appendix3.tex:150]: Run through `latex-parser`

**Comment by Xintong-Li** [comment]
Update: 

1. Modified exercise 3. Now it is a question related to basis (including orientation) and determinant.
2. Modified exercise 6. Now it is a T/F question.
3. Added a warning box.
4. Added subentries of `Determinant` in index: `$2\times 2$ formula`, `$3\times 3$ formula`, `$3\times 3$ diagonal trick`, and `$3\times 3$ cofactor expansion`.

Comment:
Warning box is not something predefined. Thus, I created a warning box using `emphbox`. The problem is, this warning box appears in green, which seems to not be a warning. Perhaps I need to find a way to temparily change the color of this emphbox.

TODO:
Add a geometrical proof of the 2-by-2 formula.

**Comment by Xintong-Li** [comment]
**Update:** Added a geometrical proof of the 2-by-2 formula.

**Comment. ** I am trying to give a proof where no square roots or triangle ratios are involved. However, I met the problem that I cannot find a "unified" way of proving the theorem.  Thus, I seperated the proof into several cases.

In addition, I am not quite familiar with the package `Tikz`, so I just modified the diagram in another section.

File: book/linearalgebra.tex [hunk 13]
```diff
@@ -8925,6 +8924,19 @@ \section*{Eigenvectors}
 	\input{modules/appendix2-exercises.tex}
 \end{appendix}
 
+\begin{appendix}
+	\Title{Determinants of Order 2 and 3}
```
Comment (siefkenj) [inline book/linearalgebra.tex:8928]: -> `Formulas for 2x2 and 3x3 Determinants`

File: book/linearalgebra.tex [hunk 14]
```diff
@@ -8925,6 +8924,19 @@ \section*{Eigenvectors}
...
+	In this appendix you will learn:
+\begin{itemize}
+  \item A practical formula for \(2\times 2\) determinants.
```
Comment (siefkenj) [inline book/linearalgebra.tex:8931]: `\(..\)` -> `$...$`

File: book/modules/appendix3-exercises.tex [hunk 15]
```diff
@@ -0,0 +1,74 @@
+\begin{exercises}
+	\begin{problist}
+		\prob For each matrix given below, calculate its determinant by (1) row reduction, and (2) using the $2\times 2$ determinant formula.
```
Comment (siefkenj) [inline book/modules/appendix3-exercises.tex:3]: Format with the online formatter.

File: book/modules/appendix3.tex [hunk 16]
```diff
@@ -0,0 +1,476 @@
...
+\Heading{Calculating \(2\times 2\) Determinants} For a $2\times 2$ matrix, once we
+know its four entries, we can calculate its determinant directly.
+\begin{theorem}[$2\times 2$ determinant formula]\index{Determinant!$2\times 2$ formula}
+	Let $M=\mat{a & b \\ c & d}$. Then
+	\[
+		\det(M)=ad-bc.
+	\]
+\end{theorem}
+\begin{proof}
+Let $\mathcal{T}_{M}$ be the matrix multiplication mapping induced by \(M\), namely \[\mathcal{T}_{M}:\mathbb{R}^2\rightarrow \mathbb{R}^2,\quad \text{ where }\mathcal{T}(\vec x)=M\vec x.\]
```
Comment (siefkenj) [inline book/modules/appendix3.tex:18]: Format with the online formatter.

**Comment by siefkenj** [comment]
Any figure you're not sure how to do, you can mark with `XXX Figure` and a description of the figure and we can fill it in later.

File: book/modules/appendix3.tex [hunk 17]
```diff
@@ -0,0 +1,476 @@
...
+
+If $\Set{\matc{a \\ c},\matc{b \\ d}}$ is linearly dependent, then $\dim \Range(\mathcal{T}_M)<2$, then $\mathcal {T}_{{M}}(C_2)$ is either a point or a one-dimensional object. In either case it has signed area $0$, so $\det(\mathcal{T}_{M})=0$. Thus, $\det(M)=0$. Now, the fact that $\Set{\matc{a \\ c},\matc{b \\ d}}$ is linearly dependent implies that there exists a scalar $\lambda$ such that $\matc{a \\ c}=\lambda\matc{b \\ d}$. Then $ad-bc=(\lambda b)d-b(\lambda c)=0$. Thus, $\det(M)=ad-bc$ in the case when column vectors of $M$ are linearly dependent.
```
Comment (JackC92) [inline book/modules/appendix3.tex:20]: You can use `\Set*{}` for the enlarged curly brackets.

Comment (siefkenj) [reply book/modules/appendix3.tex:20]: `C_2` is the unit 2-cube. But it would be good to remind the reader of that.

**Comment by Xintong-Li** [comment]
Update: Formatting

Formatted the proof using LaTeX Parser.

Change all manually crafted set braces to `\Set*{}`

**Pull request merged**